browser/ipfs/content_browser_client_helper.cc

```diff
@@ -45,25 +45,46 @@ bool IsIPFSLocalGateway(content::BrowserContext* browser_context) {
 
 namespace ipfs {
 
+// static
+bool ContentBrowserClientHelper::HandleIPFSURLRewrite(GURL* url,
+    content::BrowserContext* browser_context) {
+  if (!IsIPFSDisabled(browser_context) &&
+      // When it's not the local gateway we don't want to show a ipfs:// URL.
+      // We instead will translate the URL later in LoadOrLaunchIPFSURL.
+      IsIPFSLocalGateway(browser_context) &&
+      (url->SchemeIs(kIPFSScheme) || url->SchemeIs(kIPNSScheme))) {
+    return TranslateIPFSURI(*url, url, true);
+  }
+
+  return false;
+}
+
 // static
 bool ContentBrowserClientHelper::HandleIPFSURLReverseRewrite(GURL* url,
     content::BrowserContext* browser_context) {
   return false;
 }
 
+// static
+bool ContentBrowserClientHelper::ShouldNavigateIPFSURI(const GURL& url,
+    GURL* new_url, content::BrowserContext* browser_context) {
+  *new_url = url;
+  bool isIPFSScheme = url.SchemeIs(kIPFSScheme) || url.SchemeIs(kIPNSScheme);
+  return !IsIPFSDisabled(browser_context) && (!isIPFSScheme ||
+      TranslateIPFSURI(url, new_url, IsIPFSLocalGateway(browser_context)));
+}
+
 // static
 void ContentBrowserClientHelper::LoadOrLaunchIPFSURL(
-    const GURL& url,
+    GURL url,
```

Comment:
I think it would be better to keep this const and make a mutable copy inside the function, unless there's some other reason to do it this way.

Reply:
thanks, applying to https://github.com/brave/brave-core/pull/6697